sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -909,4 +910,17 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
       }
     }
   }
+
+  test("SPARK-34613: Fix view does not capture disable hint config") {
```

Comment:
does this test fail before? you did not change the config after the view creation.

Reply:
Yes it will fail. View just use the stored config to do anlysis. Before this fix, the config of disable hint can't be captured, then we would always use the default value `false`.

Reply:
But when we read the view, we are still under `withSQLConf(DISABLE_HINTS.key -> "true")`, isn't it?

Reply:
We don't use the `SQLConf.get` to do the view analysis, instead we use the `new SQLConf()`. So we always use the default value `false`. See some related code in `View` below
```
def effectiveSQLConf(configs: Map[String, String], isTempView: Boolean): SQLConf = {
  val activeConf = SQLConf.get
  // For temporary view, we always use captured sql configs
  if (activeConf.useCurrentSQLConfigsForView && !isTempView) return activeConf

  val sqlConf = new SQLConf()
  for ((k, v) <- configs) {
    sqlConf.settings.put(k, v)
  }
  sqlConf
}
```

Reply:
ah I see, thanks for the explanation!

Reply:
Maybe need to make description more clear next time.